### ISSUE TITLE
Build failed on latest commit of branch `OpenSSL_1_1_1-stable`

### ISSUE THREAD

**Issue by IceCodeNew** [openssl/openssl#13896]
```
$ ./config --prefix="/build_root/.openssl" --release no-deprecated no-shared no-dtls1-method no-tls1_1-method no-sm2 no-sm3 no-sm4 no-rc2 no-rc4 threads CFLAGS="$CFLAGS -fPIC" CXXFLAGS="$CXXFLAGS -fPIC" LDFLAGS='-fuse-ld=lld'
$ make -j "$(nproc)" CFLAGS="$CFLAGS -fPIE -Wl,-pie" CXXFLAGS="$CXXFLAGS -fPIE -Wl,-pie"

${LDCMD:-gcc} -pthread -m64 -Wa,--noexecstack -O2 -pipe -D_FORTIFY_SOURCE=2 -fexceptions -fstack-clash-protection -fstack-protector-strong -g -grecord-gcc-switches -Wl,-z,noexecstack,-z,relro,-z,now,-z,defs -Wl,--icf=all -fPIE -Wl,-pie -L. -fuse-ld=lld \ -o test/verify_extra_test test/verify_extra_test.o \ test/libtestutil.a -lcrypto -ldl -pthread
ld.lld: error: undefined symbol: X509_STORE_CTX_trusted_stack >>> referenced by verify_extra_test.c:203 (test/verify_extra_test.c:203) >>> test/verify_extra_test.o:(test_self_signed) >>> referenced by verify_extra_test.c:203 (test/verify_extra_test.c:203) >>> test/verify_extra_test.o:(test_self_signed)
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:9139: test/verify_extra_test] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build_root/openssl-1.1.1j-dev'
make: *** [Makefile:174: all] Error 2
```

You can replicate this built process with Dockerfile:
https://github.com/IceCodeNew/builder_image_x86_64-linux/blob/3679c1b123d815839b3a9eea0666fd8298bb4795/ubuntu_builder.Dockerfile#L84-L101

**Comment by IceCodeNew** [comment]
Related commit: https://github.com/openssl/openssl/commit/76ed0c0ad119569f6e6f6c96b27b76d3b110413b#diff-98a5595d664cdf58a493038ef0ecfe301f436e57bf09887270911ee11bb5bfbbR203

**Comment by t8m** [comment]
That's weird - I do not see the failure in the CI. Also the X509_STORE_CTX_trusted_stack is a macro defined in x509_vfy.h which should be included from x509.h. Are you sure you aren't somehow messing the build by not having correct paths for includes?

**Comment by IceCodeNew** [comment]
> Are you sure you aren't somehow messing the build by not having correct paths for includes?

I don't think so. That Dockerfile has served the base image building CI for months. And the CI only failed for the latest five days.

**Comment by t8m** [comment]
Can you please paste the lines from the build log where the verify_extra_test.c is compiled.

**Comment by IceCodeNew** [comment]
> Can you please paste the lines from the build log where the verify_extra_test.c is compiled.

```
gcc -Iinclude -pthread -m64 -Wa,--noexecstack -O2 -pipe -D_FORTIFY_SOURCE=2 -fexceptions -fstack-clash-protection -fstack-protector-strong -g -grecord-gcc-switches -Wl,-z,noexecstack,-z,relro,-z,now,-z,defs -Wl,--icf=all -fPIE -Wl,-pie -DNDEBUG -DOPENSSL_API_COMPAT=0x10100000L -MMD -MF test/verify_extra_test.d.tmp -MT test/verify_extra_test.o -c -o test/verify_extra_test.o test/verify_extra_test.c

test/verify_extra_test.c: In function 'test_self_signed':
test/verify_extra_test.c:203:5: warning: implicit declaration of function 'X509_STORE_CTX_trusted_stack'; did you mean 'X509_STORE_CTX_set0_trusted_stack'? [-Wimplicit-function-declaration] 203 | X509_STORE_CTX_trusted_stack(ctx, trusted); | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~ | X509_STORE_CTX_set0_trusted_stack
```
![image](https://user-images.githubusercontent.com/32576256/105034051-ebf00480-5a93-11eb-9163-f6d27cb9b742.png)

**Comment by t8m** [comment]
It's the no-deprecated.

**Comment by t8m** [comment]
Fix was merged to the 1.1.1 branch.
